modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/http/netty4/Netty4IncrementalRequestHandlingIT.java

```diff
@@ -175,12 +177,21 @@ public void testClientConnectionCloseMidStream() throws Exception {
             var handler = ctx.awaitRestChannelAccepted(opaqueId);
             assertBusy(() -> assertNotNull(handler.stream.buf()));
 
-            // enable auto-read to receive channel close event
-            handler.stream.channel().config().setAutoRead(true);
             assertFalse(handler.streamClosed);
 
-            // terminate connection and wait resources are released
-            ctx.clientChannel.close();
+            final List<Runnable> runnables = new ArrayList<>(
```

Comment:
Do you need shuffle? Each method of the list will be dispatched in different event loop: close() on client's, next() on servers's. The order already non deterministic

Reply:
You are right. No need for shuffle. Pushed 456e42b05b2fa8b0f5cc4eb0e6b65e3e1bc84e85 thanks